internal/statemachine/classic_states.go

```diff
@@ -581,10 +581,46 @@ func (stateMachine *StateMachine) installPackages() error {
 	var installPackagesCmds []*exec.Cmd
 
 	// mount some necessary partitions from the host in the chroot
-	mounts := []string{"/dev", "/proc", "/sys"}
+	type mountPoint struct {
+		dest     string
+		fromHost bool
+	}
+	mountPoints := []mountPoint{
+		{
+			dest:     "/dev",
+			fromHost: true,
+		},
+		{
+			dest:     "/proc",
+			fromHost: true,
+		},
+		{
+			dest:     "/sys",
+			fromHost: true,
+		},
+		{
+			dest:     "/run",
+			fromHost: false,
+		},
+	}
+
 	var umounts []*exec.Cmd
-	for _, mount := range mounts {
-		mountCmd, umountCmd := mountFromHost(stateMachine.tempDirs.chroot, mount)
+	for _, mount := range mountPoints {
+		var mountCmd, umountCmd *exec.Cmd
+		var err error
+		if mount.fromHost {
+			mountCmd, umountCmd = mountFromHost(stateMachine.tempDirs.chroot, mount.dest)
+		} else {
+			mountCmd, umountCmd, err = mountNewFS(stateMachine.tempDirs.chroot,
+				stateMachine.tempDirs.scratch,
+				mount.dest,
+			)
+			if err != nil {
+				return fmt.Errorf("Error creating temporary directory for mountpoint: \"%s\"",
+					err.Error())
```

Comment:
Conceptual nitpick: I know this is basically what the error is supposed to mean, but the error we return here is for `mountNewFS` which does more than just create a temporary directory. I know right now it's the only place where the function can fail, but it feels a bit confusing. I'd report here something in the likes: error mounting temporary directory for mountpoint x: err.Error() (which should contain the error message for failed directory creation.